src/trafficcontroller/dopplerproxy/doppler_proxy.go

```diff
@@ -137,13 +137,28 @@ func (proxy *Proxy) serveAppLogs(paths []string, writer http.ResponseWriter, req
 		return
 	}
 
-	authorized, err := proxy.logAuthorize(authToken, appId, proxy.logger)
+	status, err := proxy.logAuthorize(authToken, appId, proxy.logger)
+	if status != http.StatusOK {
+		proxy.logger.Warndf(map[string]interface{}{
+			"status": status,
+			"err":    err,
+		}, "auth token not authorized to access appId [%s].", appId)
+		switch status {
+		case http.StatusUnauthorized:
+			writer.WriteHeader(status)
+			writer.Header().Set("WWW-Authenticate", "Basic")
+			fmt.Fprintf(writer, "You are not authorized.")
+		case http.StatusForbidden:
+			status = http.StatusNotFound
+			fallthrough
+		case http.StatusNotFound:
+			writer.WriteHeader(status)
+			fmt.Fprintf(writer, "App ID is not found.")
+		default:
+			writer.WriteHeader(status)
```

Comment:
Why would you expect a 201? If you cannot explain a 201 on a GET, I would assume it's an error. The 500 is there to signify that some unexpected issue occurred not to just return things just because that is what came back.